src/events/tls.py

```diff
@@ -263,11 +264,17 @@ def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
             {"certificate": event.certificate, "ca-cert": event.ca, "ca": ""}
         )
 
-        self.dependent.tls_manager.set_server_key()
-        self.dependent.tls_manager.set_ca()
-        self.dependent.tls_manager.set_certificate()
-        self.dependent.tls_manager.set_truststore()
-        self.dependent.tls_manager.set_keystore()
+        self.charm.broker.tls_manager.set_server_key()
+        self.charm.broker.tls_manager.set_ca()
+        self.charm.broker.tls_manager.set_certificate()
+        self.charm.broker.tls_manager.set_truststore()
+        self.charm.broker.tls_manager.set_keystore()
+
+        self.charm.balancer.tls_manager.set_server_key()
+        self.charm.balancer.tls_manager.set_ca()
+        self.charm.balancer.tls_manager.set_certificate()
+        self.charm.balancer.tls_manager.set_truststore()
+        self.charm.balancer.tls_manager.set_keystore()
```

Comment:
**todo:** For my eyeballs, could we loop this?
```python
for dependent in ["broker", "balancer"]:
    getattr(self.charm, dependent).tls_manager.set_server_key()
    getattr(self.charm, dependent).tls_manager.set_ca()
    ...
```

Reply:
lol yes. this wasn't even its final form. kraft with 15 lines of the same would've been epic.